app/models/campaign_pages_widget.rb

```diff
@@ -1,12 +1,11 @@
 class CampaignPagesWidget < ActiveRecord::Base
 
-  belongs_to :campaign_page
+  belongs_to :campaign_page, inverse_of: :campaign_pages_widgets
   belongs_to :widget_type
   has_one :actionkit_page
 
   validates_presence_of :content, :page_display_order, :campaign_page_id, :widget_type_id
   # validates that there are not two widgets with exactly same content on the same campaign page
```

Comment:
This is the campaign_pages_widgets belongs_to campaign_page combination - this will have to stay. The inverse is specified explicitly to make sure nested attributes work all right. I guess you mean the campaign pages to templates connection?